tools/chpl-language-server/src/chpl-language-server.py

```diff
@@ -291,11 +296,53 @@ class ResolvedPair:
     resolved_to: NodeAndRange
 
 
+class ContextContainer:
+    context: chapel.core.Context
+    file_paths: List[str]
+    module_paths: List[str]
+    file_infos: List['FileInfo']
+
+    def _get_configuration(self, path, project_root):
+        cls_config = os.path.join(project_root, ".cls-info.json")
+        self.module_paths = []
+        self.file_paths = [path]
+        if os.path.exists(cls_config):
+            with open(cls_config) as f:
+                commands = json.load(f)
+                if path in commands:
+                    self.module_paths = commands[path][0]["module_dirs"]
+                    self.file_paths = commands[path][0]["files"]
+        self.context.set_module_paths(self.module_paths, self.file_paths)
+
+    def __init__(self, file: str, project_root: str):
+        self.context = chapel.core.Context()
+        self.file_paths = []
+        self.module_paths = []
+        self.file_infos = []
+
+        self._get_configuration(file, project_root)
```

Comment:
Rather than threading through the project root, it would be better separation to pass in a config dict and do the file loading elsewhere.

I am also not a fan of the file loading logic you have, namely the `[0]`. I would prefer something that handles error cases. What if there is more than one? What if there is none?